firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerformance.java

```diff
@@ -194,6 +195,7 @@ public static FirebasePerformance getInstance() {
     this.configResolver = configResolver;
     this.configResolver.setMetadataBundle(mMetadataBundle);
     this.configResolver.setApplicationContext(appContext);
+    // TODO (Should this be done after the register call to firebase sessions?)
     sessionManager.setApplicationContext(appContext);
```

Comment:
I guess what is the impact of calling this?

Reply:
The actual impact here is that the sessionID relevant to app start happens when the applicationContext is set. With SessionsRegistration done later after setting applicationContext, it would happen that app start trace might not have a sessionID associated with it.

Reply:
I see so in the case of a crash or something interrupting it